src/FixedDataTableRow.js

```diff
@@ -189,16 +192,29 @@ var FixedDataTableRowImpl = React.createClass({
     return width;
   },
 
-  _renderColumnsShadow(/*number*/ left) /*?object*/ {
-    if (left > 0) {
+  _renderColumnsLeftShadow(/*number*/ left) /*?object*/ {
+    var className = cx({
+      'fixedDataTableRowLayout/fixedColumnsDivider': left > 0,
+       'fixedDataTableRowLayout/columnsShadow': this.props.scrollLeft > 0,
+       'public/fixedDataTableRow/fixedColumnsDivider': left > 0,
+       'public/fixedDataTableRow/columnsShadow': this.props.scrollLeft > 0,
+     });
+     var style = {
+       left: left,
+       height: this.props.height
+     };
+     return <div className={className} style={style} />;
+   },
+
+  _renderColumnsRightShadow(/*number*/ totalWidth) /*?object*/ {
+    if (this.props.scrollLeft + this.props.width + 0.5 < totalWidth) {
```

Comment:
Changed to use 'Math.ceil' to roundup the value so it's more readable. There is a chance the shadow doesn't show nicely at an edgy case otherwise.